src/components/Tasks/Sidebar.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+
+import { useSelector, useDispatch } from 'react-redux';
+
+import List from '@material-ui/core/List';
+import ListSubheader from '@material-ui/core/ListSubheader';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemText from '@material-ui/core/ListItemText';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import { IconProps } from '@material-ui/core/Icon';
+import LabelIcon from '@material-ui/icons/LabelOutlined';
+
+import { setSettings } from '../../store/actions';
+import { StoreState } from '../../store';
+
+interface ListItemPropsType {
+  name: string | null;
+  Icon?: React.ComponentType<IconProps>;
+  primaryText: string;
+  secondaryText?: string;
+}
+
+function SidebarListItem(props: ListItemPropsType) {
+  const { name, Icon, primaryText, secondaryText } = props;
+  const dispatch = useDispatch();
+  const taskSettings = useSelector((state: StoreState) => state.settings.taskSettings);
+  const { filterBy } = taskSettings;
+
+  const handleClick = () => {
+    dispatch(setSettings({ taskSettings: { ...taskSettings, filterBy: name } }));
+  };
+
+  return (
+    <ListItem
+      button
+      onClick={handleClick}
+      selected={name === filterBy}
+    >
+      {Icon ?
+        <ListItemIcon>
+          <Icon fontSize="small" />
+        </ListItemIcon> :
+        ''
+      }
+      <ListItemText primary={primaryText} secondary={secondaryText || ''} />
+    </ListItem>
+  );
+}
+
+export default function Sidebar(props: { tags: Map<string, number> }) {
+  const { tags } = props;
+
+  const tagsList = Array.from(tags, ([tag, amount]) => (
+    <SidebarListItem
+      key={tag}
+      name={`tag ${tag}`}
```

Comment:
Horrible bike-shedding, I know, but could you please change this to `tag:${tag}`? So have a colon as the delimiter? I know it's just my personal preference, but using space as the delimiter fields very weird for me. I was trying to not mention it (and I wasn't going it if I had no other comments), but since I'm already commenting... :)

Reply:
That makes sense. I was also debating whether I should use a colon or something for the delimiter.